lib/home/schedule/src/schedule_editor.dart

```diff
@@ -1,14 +1,309 @@
+import 'package:intl/intl.dart';
+import 'package:table_calendar/table_calendar.dart';
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:thc/home/users/src/all_users.dart';
 import 'package:thc/the_good_stuff.dart';
-import 'package:thc/utils/widgets/placeholders.dart';
 
-class ScheduleEditor extends StatelessWidget {
+class ScheduleEditor extends StatefulWidget {
   const ScheduleEditor({super.key});
 
+  @override
+  State<ScheduleEditor> createState() => _ScheduleEditorState();
+}
+
+class _ScheduleEditorState extends State<ScheduleEditor> {
+  final GlobalKey<FormState> _formKey = GlobalKey<FormState>();
+  final TextEditingController _event = TextEditingController();
+  final TextEditingController _startDate = TextEditingController();
+  final TextEditingController _endDate = TextEditingController();
+  final TextEditingController _startTime = TextEditingController();
+  final TextEditingController _endTime = TextEditingController();
+
+  Director? director;
+
+  DateTime _selectedDay = DateTime.now();
+  Map<DateTime, List<dynamic>> _events = {};
+  CalendarFormat _calendarFormat = CalendarFormat.month;
+  DateTime _focusedDay = DateTime.now();
+
+  final eventsCollection = FirebaseFirestore.instance.collection('events');
```

Comment:
Instead of `events`, I believe we're storing the scheduled streams in [Firebase](https://console.firebase.google.com/u/2/project/the-heart-center-app/firestore/databases/-default-/) under the name `scheduled_streams`.